pkg/syncer/volume_health_reconciler.go

```diff
@@ -209,55 +308,62 @@ func (rc *volumeHealthReconciler) syncPVC(key string) error {
 	log.Infof("syncPVC: Found Supervisor Cluster PVC: %s/%s", svcPVC.Namespace, svcPVC.Name)
 
 	// The input PVC is a Supervisor Cluster PVC
-	// Find Tanzu Kubernetes Grid PV
-	// Find Tanzu Kubernetes Grid PVC accordingly and update it
-	tkgPV, err := rc.findTKGPVforSupervisorPVC(ctx, svcPVC)
+	// Find list of Tanzu Kubernetes Grid PVs referencing this PVC
+	// Find corresponding Tanzu Kubernetes Grid PVCs and update them
+	tkgPVList, err := rc.findTKGPVforSupervisorPVC(ctx, svcPVC)
 	if err != nil {
 		return err
 	}
-	if tkgPV == nil {
+	if tkgPVList == nil {
+		// This is a workaround for statically provisioned volumes.
+		// For static provisioning, because the SVC PVC is created first, when we get the event in TKG,
+		// there is no PVC in TKG yet, so it gets dropped from the queue.
+		// When a new PVC is created in TKG to bind to a new PV which points to the existing SVC PVC,
+		// it won't get the volume health annotation because the SVC PVC is already removed from the queue.
+		// By returning  an error, the reconciler will add the SVC PVC back to the queue.  It will be retried
```

Comment:
You may want to add that this is a temporary workaround to get the functionality in at the cost of performance. The performance issue is - volume health reconciler in PVCSI continously keeps processing SV PVC event that is not being used in TKG cluster.